### ISSUE TITLE
Datatable sort loses inputvalues

### ISSUE THREAD

**Issue by andyfinch** [primefaces/primefaces#829]
Using a datatable which includes columns with any input fields loses any inputted value on sort.  I believe this is to do with the skipChildren param.  However regardless of how I set that at the datatable level skipChildren is always true on sort.  This behaviour doesn't happen when you filter and all inputted values are retained.

I am using Primefaces 5.2.12

**Comment by kukel** [comment]
show an mcve...

**Comment by andyfinch** [comment]
```
<p:dataTable value="#{myBean.myListOfStrings}" var="stringVar">
        <p:column headerText="myInputColumn" sortBy="#{stringVar}">
            <h:inputText value="#{stringVar}"/>
        </p:column>
  </p:dataTable>

private List<String> myListOfStrings = Arrays.asList(null,null,null,null,null);

public List<String> getMyListOfStrings() {
    return myListOfStrings;
}

public void setMyListOfStrings(List<String> myListOfStrings) {
    this.myListOfStrings = myListOfStrings;
}
```

Run and enter value in one of the input fields. Click sort and value has gone.

I have worked around this with a javascript hack:

```
<p:ajax event="sort" onstart="removeSkipChildrenForSorting();"/>

function removeSkipChildrenForSorting()
{
    $(PrimeFaces.ajax.Queue.peek().ext.params).each(function(){
        if ( this.name.indexOf('_skipChildren') != -1 )
        {
            this.name = '_skipChildrenRemoved';
        }       
    });        
}
```

**Comment by cghislai** [comment]
I confirm this on primefaces 5.3.

For the sort event, `_skipChildren` is set to true, regardless of the value set in the p:dataTable parameters.

**Comment by kukel** [comment]
If I populate fields and there is no ajax, I do not expect them to be submitted when paging (nor with filtering so if that happens there, **that** is a bug in my opinion. Use an ajax tag if you want fields to be automatically submitted... or use a save button

**Comment by andyfinch** [comment]
`<p:ajax event = 'sort' process...../>`  makes no difference.  Not sure if it is a bug when no filtering/paging p ajax is present however I would be happy to control processing using the ajax events.  However if it is already processing without these ajax events (filtering/paging) then you could potentially break a lot of existing tables.

**Comment by kukel** [comment]
Yes, backwards compatibility is an issue... But also for **enabling** submitting if it is now **not** submitting on sort. So if 'fixed' and breaking backwards compatibility, it should only be fixed for 6.0

**Comment by cagataycivici** [comment]
This is now supported, set p:ajax event="sort" skipChildren="false".

https://github.com/primefaces/primefaces/issues/1183

Also check out

https://github.com/primefaces/primefaces/issues/1183

That enables preventing sort if an input is invalid inside.
